lib/libcompat.h

```diff
@@ -48,8 +48,10 @@
 
 #if GCC_VERSION_AT_LEAST(2,95,3)
 #define CK_ATTRIBUTE_UNUSED __attribute__ ((unused))
+#define CK_ATTRIBUTE_FORMAT(a, b, c) __attribute__ ((format (a, b, c)))
```

Comment:
What this attribute introduced in GCC 2.95.3? I could not find a reference for that. Do you know when it was introduced? 

Or, should there be a configure check for it? Not sure what it would be for CMake, but this may work for autotools:

https://www.gnu.org/software/autoconf-archive/ax_gcc_func_attribute.html

Reply:
GNU 2.95.3 is the oldest version "supported" by GNU. I don't know what their official view is but 2.95.3 is the oldest version with documentation on GCC's pages, and __attribute__ ((format)) is supported by 2.95.3. All docs: https://gcc.gnu.org/onlinedocs/
Regarding __attribute__ ((format)): https://gcc.gnu.org/onlinedocs/gcc-2.95.3/gcc_4.html#SEC84


I would not go for a configure check because this is compiler specific thing and can be checked with compiler's versions.

Reply:
Actually, thinking about it, it would be good if there was a way to check the compiler's support for *attribute ((format))* in the build system, both Autotools and CMake. After all, Clang probably supports all the GNU C originated function attributes.

I think this should go to TODO...

Reply:
Really I'm concerned about compilers other than gcc and clang. I'd suspect that the way it is now the other compilers will not reach the format attribute as they would not claim to be GCC. 

A TODO is fine

Reply:
Yes. Other compilers would get nothing instead of "__attribute__ ((format))" when macro CK_ATTRIBUTE_FORMAT is opened.